terraform/deployments/govuk-publishing-platform/signon_bearer_tokens.tf

```diff
@@ -25,65 +25,89 @@ locals {
   api_user_prefix = local.is_default_workspace ? null : local.workspace
   signon_api_user = {
-    content_store       = join("-", compact([local.api_user_prefix, "content-store@${var.publishing_service_domain}"]))
-    draft_content_store = join("-", compact([local.api_user_prefix, "draft-content-store@${var.publishing_service_domain}"]))
-    frontend            = join("-", compact([local.api_user_prefix, "frontend@${var.publishing_service_domain}"]))
-    publisher           = join("-", compact([local.api_user_prefix, "publisher@${var.publishing_service_domain}"]))
-    publishing_api      = join("-", compact([local.api_user_prefix, "publishing-api@${var.publishing_service_domain}"]))
+    content_store = {
+      app_name = "content-store"
+      email    = join("-", compact([local.api_user_prefix, "content-store@${var.publishing_service_domain}"]))
+    }
+    draft_content_store = {
+      app_name = "draft-content-store"
+      email    = join("-", compact([local.api_user_prefix, "draft-content-store@${var.publishing_service_domain}"]))
+    }
+    frontend = {
+      app_name = "frontend"
+      email    = join("-", compact([local.api_user_prefix, "frontend@${var.publishing_service_domain}"]))
+    }
+    publisher = {
+      app_name = "publisher"
+      email    = join("-", compact([local.api_user_prefix, "publisher@${var.publishing_service_domain}"]))
+    }
+    publishing_api = {
+      app_name = "publishing-api"
+      email    = join("-", compact([local.api_user_prefix, "publishing-api@${var.publishing_service_domain}"]))
+    }
   }
 
   signon_bearer_tokens = {
     cs_to_pub_api = {
-      api_user = local.signon_api_user.content_store
-      app      = local.signon_app.publishing_api.name
+      api_user   = local.signon_api_user.content_store.email
+      app        = local.signon_app.publishing_api.name
+      client_app = local.signon_api_user.content_store.app_name
     }
 
     cs_to_router_api = {
-      api_user = local.signon_api_user.content_store
-      app      = local.signon_app.router_api.name
+      api_user   = local.signon_api_user.content_store.email
+      app        = local.signon_app.router_api.name
+      client_app = local.signon_api_user.content_store.app_name
     }
 
     dcs_to_pub_api = {
-      api_user = local.signon_api_user.draft_content_store
-      app      = local.signon_app.publishing_api.name
+      api_user   = local.signon_api_user.draft_content_store.email
+      app        = local.signon_app.publishing_api.name
+      client_app = local.signon_api_user.draft_content_store.app_name
     }
 
     dcs_to_draft_router_api = {
-      api_user = local.signon_api_user.draft_content_store
-      app      = local.signon_app.draft_router_api.name
+      api_user   = local.signon_api_user.draft_content_store.email
+      app        = local.signon_app.draft_router_api.name
+      client_app = local.signon_api_user.draft_content_store.app_name
     }
 
     pub_to_pub_api = {
```

Comment:
Are we potentially missing draft Publisher to Publishing API?

Reply:
Nah, there's no draft publisher app, we only have draft instances on the frontend side of things https://docs.publishing.service.gov.uk/manual/content-preview.html#design